### PR TITLE
Edit QoS SDK for xplat c++

### PR DESCRIPTION
**Pull request by yzhao174**
Edit QoS SDK for c++. Now it returns a sorted list based on latency and send metrics to PlayStreamv2

### CODE REVIEW

File: targets/XPlatCppSdk/source/code/include/playfab/QoS/QoS.h [hunk 1]
```diff
@@ -23,5 +23,14 @@ namespace PlayFab
 
         // Number of times a server is pinged
         constexpr int NUM_OF_PING_ITERATIONS = 3;
+
+        enum QoSErrorCode
```
Comment (Vicodex) [inline targets/XPlatCppSdk/source/code/include/playfab/QoS/QoS.h]: >QoSErrorCode [](start = 13, length = 12)

Can we please add a comment for this enum explaining that these are possible error code values for QoSResult.erroCode?

File: targets/XPlatCppSdk/source/code/source/playfab/QoS/PlayFabQoSApi.cpp [hunk 2]
```diff
@@ -30,6 +94,12 @@ namespace PlayFab
         QoSResult PlayFabQoSApi::GetQoSResult(unsigned int numThreads, unsigned int timeoutMs)
         {
             QoSResult result(move(GetResult(numThreads, timeoutMs)));
+
+            if (result.errorCode != QoSErrorCode::Success)
+            {
+                return result;
```
Comment (Vicodex) [inline targets/XPlatCppSdk/source/code/source/playfab/QoS/PlayFabQoSApi.cpp]: >return result; [](start = 16, length = 14)

Is it intentional that we don't want to send failures to PlayFab?

Comment (yzhao174) [reply targets/XPlatCppSdk/source/code/source/playfab/QoS/PlayFabQoSApi.cpp]: Yeah, this is caused by either player not logged in or PlayFab doesn't reply with QosServers list. We shouldn't spam the event stream with those failures.

File: targets/XPlatCppSdk/source/code/include/playfab/QoS/QoS.h (hunk 1 above)
Comment (yzhao174) [reply targets/XPlatCppSdk/source/code/include/playfab/QoS/QoS.h]: I will add commnents

**Comment by pgilmorepf** [comment]
Landing this will be delayed until the XPlat automated testing is brought up to date, in line with the other SDKs. Apologies, we're working on it now.

**Comment by Vicodex** [comment]
retest

**Pull request merged**